lib/agent.js

```diff
@@ -214,9 +214,15 @@ Agent.prototype._handle = function handle(msg, rsinfo, outSocket) {
   }
 
   if (req.response) {
-    // it is an observe request
-    // and we are already streaming
-    return req.response.append(packet)
+    if (req.response.append) {
+      // it is an observe request
+      // and we are already streaming
+      return req.response.append(packet)
+    } else {
+      // TODO There is a previous response but is not an ObserveStream !
+      return
+    }
```

Comment:
The CON response is already ACKed when we reach point of detecting the duplicate. If you look at Agent._handle, the first thing done is to send an ACK if the response is CON.